### DISCUSSION TITLE
GET me

### DISCUSSION THREAD

**Discussion by jor70** [noweh/twitter-api-v2-php#41] [Q&A] [answered]
Hi, thanks for the api, works great for posting with V2  
Can you please provide an example how to use GET ME endpoint,  due to this is one (along post and delete) is the only access we have to access our own tweets
thks

**Comment by noweh** [comment]
Hi and thanks for your message, this endpoint is not managed at the moment :(.

It would be necessary to update the processing on this file: https://github.com/noweh/twitter-api-v2-php/blob/master/src/UserLookup.php

**Reply by jor70** [reply]
it is in the todo list ?  :)

**Reply by noweh** [reply, marked as answer]
Several other topics for the moment... In any case, I took note of it :)

**Reply by noweh** [reply]
I have implemented the handling of this route in a new tag (3.4.0). However, its functionality appears to be quite limited at the moment. It seems that we can only retrieve the pinned tweet. For more details, please refer to: [https://github.com/noweh/twitter-api-v2-php#find-me](https://github.com/noweh/twitter-api-v2-php#find-me).

I've attempted to provide all possible parameters to obtain the most detailed response possible.

**Reply by octaviotron** [reply]
Hello. I am using the library with free access V2 API account. 

Some errors are raised using GETme due to an error because some parameters are not allowed (only in basic and pro access)

I solved this changing UserMeLookup.php in this way:
```

    protected function constructEndpoint(): string
    {
   /*
        return parent::constructEndpoint() .
            '?user.fields=created_at,description,entities,id,location,name,' .
            'profile_image_url,protected,public_metrics,url,username,verified,verified_type,withheld' .
            '&tweet.fields=attachments,author_id,context_annotations,conversation_id,created_at,edit_controls,' .
            'edit_history_tweet_ids,entities,geo,id,in_reply_to_user_id,lang,non_public_metrics,note_tweet,' .
            'organic_metrics,possibly_sensitive,public_metrics,referenced_tweets,reply_settings,source,text,withheld' .
            '&expansions=pinned_tweet_id'
        ;
   */
         return parent::constructEndpoint() .
            '?user.fields=created_at,description,id,location,name,' .
            'profile_image_url,protected,public_metrics,url,username,verified,verified_type,most_recent_tweet_id,entities&expansions=pinned_tweet_id'
        ;
    }
```

No errors now :-)

**Reply by noweh** [reply]
Thank you for this feedback ! I will look to make changes
